Adding machine output for list and repo list commands

EnriqueL8 opened the pull request:
Inspired from #230 
Fixes #53 

I think the tests can be better... Let me know what you think @kylegc

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/appsody/appsody?pullRequest=394) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/appsody/appsody?pullRequest=394) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/appsody/appsody?pullRequest=394) it.</sub>

kylegc commented on cmd/repo_list.go:
```diff
@@ -15,10 +15,18 @@
 )
 
+type repoListCommandConfig struct {
+	output string
```
Same comment to add `*RootCommandConfig`

kylegc commented on cmd/list_test.go:
```diff
@@ -74,12 +75,6 @@ func TestListV2(t *testing.T) {
-	output, _ = cmdtest.RunAppsodyCmdExec([]string{"list", "appsodyhub"}, ".")
-
-	// we expect 2 instances
-	if !(strings.Count(output, "java-microprofile") == 1) {
```
I had some issues with this test as well, and I think it needs to be reworked. However this part is not failing for me, I was getting issues with the `strings.Count(output, "nodejs ") == 2` a couple lines down. What was the reason for removing this?

EnriqueL8 replied:
I had also errors with the `"nodejs "` line, but it gets fixed when you run the whole test again because the previous test adds a new repository with a `nodejs` stack.

I removed this because the same check is performed when doing the following, just above.
```go
if !strings.Contains(output, "java-microprofile") {
	t.Error("list command should contain id 'java-microprofile'")
}
```

kylegc commented on cmd/repo.go:
```diff
@@ -302,7 +302,7 @@ func downloadIndex(url string) (*RepoIndex, error) {
 }
 
 func (index *RepoIndex) listProjects(repoName string, config *RootCommandConfig) (string, error) {
```
While you are changing this code, can you come up with better names for the two `listProjects` methods?

EnriqueL8 replied:
After looking again at these function, I believe this needs more work than just renaming. They should just be extract into a common function, since they share a lot of code.

kylegc replied:
Fair enough. Please create a separate issue to fix this :)

EnriqueL8 replied:
Just seen this! Yes, I'll create an issue to maybe refactor `repo.go`?

EnriqueL8 commented:
Thanks for the comments @kylegc! 
- I have changed the key names of JSON to be consistent with the YAML one.
- I have changed it to an array of Templates, and have added a function to convert the array to a string for printing. There is an inconsistency between JSON and YAML when outputting Boolean values using the `omitempty` flag. If a boolean is set to `False` YAML does not output it, but JSON does. This is related to adding the `default` field for the template.
- For the inconsistency between the two commands, I believe that the `repo list` command is related to precise information about the repos being used that is why the APIVersion might be of use. For `list` we just want the list of stacks available, I wonder what would be the use case to have the APIVersion there. I will change the code to add the APIVersion into the list command.

kylegc commented:
For the default field, what if you do not specify `omitempty`? Will it be displayed as false then?

EnriqueL8 commented:
> For the default field, what if you do not specify `omitempty`? Will it be displayed as false then?

Yes
